nautobot/extras/filters.py

```diff
@@ -418,55 +431,181 @@ class Meta:
 )
 
 
-class CustomFieldFilter(django_filters.Filter):
+class CustomFieldFilterMixin:
     """
-    Filter objects by the presence of a CustomFieldValue. The filter's name is used as the CustomField name.
+    Filter mixin for CustomField to handle CustomField.filter_logic setting
+    and queryset.exclude filtering specific to the JSONField where CustomField data is stored.
     """
 
     def __init__(self, custom_field, *args, **kwargs):
         self.custom_field = custom_field
-
-        if custom_field.type == CustomFieldTypeChoices.TYPE_INTEGER:
-            self.field_class = IntegerField
-        elif custom_field.type == CustomFieldTypeChoices.TYPE_BOOLEAN:
-            self.field_class = NullBooleanField
-        elif custom_field.type == CustomFieldTypeChoices.TYPE_DATE:
-            self.field_class = DateField
-
-        super().__init__(*args, **kwargs)
-
-        self.field_name = f"_custom_field_data__{self.field_name}"
-
         if custom_field.type not in EXACT_FILTER_TYPES:
             if custom_field.filter_logic == CustomFieldFilterLogicChoices.FILTER_LOOSE:
-                self.lookup_expr = "icontains"
-
-        elif custom_field.type == CustomFieldTypeChoices.TYPE_MULTISELECT:
-            # Contains handles lists within the JSON data for multi select fields
-            self.lookup_expr = "contains"
+                kwargs.setdefault("lookup_expr", "icontains")
+        super().__init__(*args, **kwargs)
+        self.field_name = f"_custom_field_data__{self.field_name}"
 
     def filter(self, qs, value):
+        if value in django_filters.constants.EMPTY_VALUES:
+            return qs
+
         if value == "null":
             return self.get_method(qs)(
                 Q(**{f"{self.field_name}__exact": None}) | Q(**{f"{self.field_name}__isnull": True})
             )
+
+        # Custom fields require special handling for exclude filtering.
+        # Return custom fields that don't match the value and null custom fields
+        if self.exclude:
+            qs_null_custom_fields = qs.filter(**{f"{self.field_name}__isnull": True}).distinct()
+            return super().filter(qs, value) | qs_null_custom_fields
+
         return super().filter(qs, value)
 
 
+class CustomFieldBooleanFilter(CustomFieldFilterMixin, django_filters.BooleanFilter):
+    """Custom field single value filter for backwards compatibility"""
+
+    pass
```

Comment:
Technically these `pass` shouldn't be needed, the above by itself is perfectly valid.
```suggestion
```